### PR TITLE
Parceler migration

### PR DESCRIPTION
**Pull request by AndyScherzinger**
migrating further parts of the api classes

### CODE REVIEW

File: scripts/hooks/pre-commit [hunk 1]
```diff
@@ -5,6 +5,7 @@ echo "Running pre-commit checks..."
 if ! ./gradlew --daemon ktlintCheck &>/dev/null; then
     echo >&2 "ktlint failed! Run ./gradlew ktlintCheck for details"
     echo >&2 "Hint: fix most lint errors with ./gradlew ktlintFormat"
+	echo >&2 "Detekt failed! See report at file://$(pwd)/app/build/reports/ktlint/ktlintMainSourceSetCheck/ktlintMainSourceSetCheck.txt"
```
Comment (AlvaroBrey) [inline scripts/hooks/pre-commit:8]: `s/Detekt/ktlint/`

The reason that I didn't include this in the original hooks in Files, and instead direct the user to run `ktlintCheck` for results, is that `ktlint` generates a separate report for each sourceset. If you have an error in the `test` sourceset, for example, it won't show up in the linked report.

Comment (AndyScherzinger) [reply scripts/hooks/pre-commit:8]: So remove it again @AlvaroBrey

Comment (AlvaroBrey) [reply scripts/hooks/pre-commit:8]: I would remove it, yeah.

Comment (AndyScherzinger) [reply scripts/hooks/pre-commit]: DONE 👍

**Review by mahibi — APPROVED** [review]

**Review by timkrueger — APPROVED** [review]
More 💙💙💙 for @AndyScherzinger :smile:

**Pull request merged**